### PR TITLE
Add writing direction to locale files

### PR DESCRIPTION
**Pull request by ghost**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Locale files do not contain writing direction

## What is the new behavior?
Locale files will contain writing direction

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
Look at first commit for actual changes. Second commit is just generated locale files

### CODE REVIEW

**Comment by doom777** [comment]
If you're wondering the purpose of this PR, it is to eventually support angular/angular-cli#16047 
Currently, creating an RTL locale build does not make the page RTL

File: packages/common/src/i18n/locale_data_api.ts [hunk 1]
```diff
@@ -562,6 +562,17 @@ export function getLocaleExtraDayPeriods(
   return getLastDefinedValue(dayPeriods, width) || [];
 }
 
+/**
+ * Retrieves the writing direction of a specified locale *
+ * @param locale A locale code for the locale format rules to use.
+ * @returns 'RTL' or 'LTR
+ * @see [Internationalization (i18n) Guide](https://angular.io/guide/i18n)
```
Comment (petebacondarwin) [inline packages/common/src/i18n/locale_data_api.ts]: will also need a `@publicApi` tag here.

File: packages/common/src/i18n/locale_data_api.ts [hunk 2]
```diff
@@ -562,6 +562,17 @@ export function getLocaleExtraDayPeriods(
   return getLastDefinedValue(dayPeriods, width) || [];
 }
 
+/**
+ * Retrieves the writing direction of a specified locale *
```
Comment (petebacondarwin) [inline packages/common/src/i18n/locale_data_api.ts]: Dangling `*` character - remove?

File: tools/gulp-tasks/cldr/extract.js [hunk 3]
```diff
@@ -498,6 +499,15 @@ function getCurrencySettings(locale, localeData) {
   return currencySettings;
 }
 
+/**
+ * Returns the writing direction for a locale
+ * @returns 'RTL' | 'LTR'
+ */
+function getDirectionality(localeData) {
+  const rtl = localeData.get('scriptMetadata/{script}/rtl');
+  return rtl === 'YES' ? 'RTL' : 'LTR';
```
Comment (petebacondarwin) [inline tools/gulp-tasks/cldr/extract.js]: From reading the HTML spec it looks like it is normal to use lowercase for the `dir` attribute. So I would recommend using the same here.

File: packages/common/src/i18n/locale_data_api.ts [hunk 4]
```diff
@@ -562,6 +562,17 @@ export function getLocaleExtraDayPeriods(
...
+ * @param locale A locale code for the locale format rules to use.
+ * @returns 'RTL' or 'LTR
+ * @see [Internationalization (i18n) Guide](https://angular.io/guide/i18n)
+ */
+export function getLocaleDirection(locale: string): 'LTR'|'RTL' {
```
Comment (petebacondarwin) [inline packages/common/src/i18n/locale_data_api.ts]: See comment about lower casing these on `getDirectionality()` function

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Thanks for this PR - generally looks good. A couple of items to look at.
We should also consider whether this should land before or after https://github.com/angular/angular/pull/33523 as they clearly conflict with each other.

I think also that since we are well into the RC now, and this has a public API change, then this will probably go into 9.1.0

**Comment by doom777** [comment]
@petebacondarwin do you suggest I rebase this change on top of #33523?

**Comment by petebacondarwin** [comment]
Let's hold off for 24 hours while we work out what release this could go in and what to do about that other PR. Cheers

**Review by petebacondarwin — APPROVED** [review]

**Comment by petebacondarwin** [comment]
I've marked this as `target: master only` as I think it might have to go into 9.1.x.
@IgorMinar do you have a view?

**Review by josephperrott — APPROVED** [review]
LGTM, for dev-infra

**Comment by doom777** [comment]
I noticed that #33523 was merged in. I guess I'll rebase this PR on top of master.

**Comment by petebacondarwin** [comment]
Thanks @doom777 - sorry I didn't get back to you earlier.

**Comment by petebacondarwin** [comment]
This is now competing with https://github.com/angular/angular/pull/33584 since they will also conflict :-/

**Comment by doom777** [comment]
@petebacondarwin true, but my PR was here first ;-)
Also, this PR is further ahead of the other one, since we got more approvals, and are merged with master.
Also, if you really insist, I can rebase this PR on top of that one also.

**Comment by petebacondarwin** [comment]
The question is really whether this can go into `master & patch` (e.g. 9.0.0) or only `master` (e.g. 9.1.0), since it is arguably a new public API and we are now in RC for 9.0.0. cc @IgorMinar / @kara ?

**Comment by doom777** [comment]
@petebacondarwin BTW, why *are* these locale files checked into source control? Why not generate them during the build process?

**Comment by petebacondarwin** [comment]
That is a good question. It has been like this for many years.
I think it is nice not to have to generate them locally when developing that Angular framework.

That being said we should look into ways to automate the update of these to some Bazel rules...

**Comment by josephperrott** [comment]
This is actually an outstanding task for the @angular/dev-infra-framework team.  To determine if continuing to store them in the repository is the best option or if they should be generated at build time.  And with whichever method, how to best do that going forward.

**Comment by AndrewKushnir** [comment]
Adding @IgorMinar to reviewers list, since there are changes to public API that need to be reviewed/approved. @IgorMinar, could you please have a look at [the first commit](https://github.com/angular/angular/pull/33556/commits/4f4c4d631461b5a4cdeec3fff9eab0b5eeb976a3) (where the API changes are)? Thank you.

**Comment by doom777** [comment]
blocked by: #33682

**Comment by doom777** [comment]
OK, we're unblocked and back ready to merge. Can we move this PR forward? @petebacondarwin @IgorMinar

**Comment by petebacondarwin** [comment]
@doom777 we need to get approval from `fw-core`, which includes @kara  and @IgorMinar. I think that given we are now well into RC, this will go into `master`, which will be released at 9.1.0

**Comment by IgorMinar** [comment]
yeah, this should be master-only and target 9.1

**Comment by doom777** [comment]
Ok, so can you approve it and into master for 9.1

**Comment by doom777** [comment]
Ok, guys. A few days passed. I understand that this is going only into master, and not patch, but can it get merged into master already? Or, because it's meant for 9.1, it doesn't get merged until a specific date?

File: packages/common/locales/en-SB.ts [hunk 5]
```diff
@@ -33,10 +35,18 @@ export default [
...
+  ['{1}, {0}', u, '{1} \'at\' {0}', u],
   ['.', ',', ';', '%', '+', '-', 'E', '×', '‰', '∞', 'NaN', ':'],
-  ['#,##0.###', '#,##0%', '¤#,##0.00', '#E0'], '$', 'Solomon Islands Dollar',
-  {'JPY': ['JP¥', '¥'], 'SBD': ['$'], 'USD': ['US$', '$']}, plural
+  ['#,##0.###', '#,##0%', '¤#,##0.00', '#E0'],
+  '$',
+  'Solomon Islands Dollar',
+  {'JPY': ['JP¥', '¥'], 'SBD': ['$'], 'USD': ['US$', '$']},
+  'ltr',
+  plural
```
Comment (petebacondarwin) [inline packages/common/locales/en-SB.ts]: I don't think this change should be in the first commit.

**Comment by petebacondarwin** [comment]
I am pinging the fw-core team to approve and run the presubmit.

**Comment by doom777** [comment]
Also the build failed for some reason. Can you rerun it?

**Comment by petebacondarwin** [comment]
It needs a rebase - that should fix the failures.

**Comment by mhevery** [comment]
[presubmit](http://test/OCL:281082299:BASE:281082333:1574097062352:52602ae)
[ivy presubmit](http://test/OCL:281082299:BASE:281082717:1574097190373:edb3bbfe)

**Comment by doom777** [comment]
Hey, so we are in `action:merge` right now, so what else needs to be done?

**Comment by matsko** [comment]
Sorry @doom777, the presubmits need to run again before merging. ETA 1-2 hours.

**Comment by matsko** [comment]
presubmit = https://test.corp.google.com/ui#id=OCL:281825168:BASE:281825201:1574373305055:9a1b9da8

ivy-presubmit = https://tap.corp.google.com/ui#id=OCL:281825168:BASE:281825333:1574373342060:bf55f5c0

**Review by IgorMinar — CHANGES_REQUESTED** [review]
Hi there, I rereviewed the PR again and noticed that `getLocaleDirection` is not reexported from `@angular/core` which means that the function is not available via the public api surface, rendering this feature inaccessible to developers (outside of our own unit tests).

Can you please update the PR with proper reexports, which will also trigger the need to update the public api guard (the CI failure will instruct you what command to run to do that).

While this PR is large, it should not conflict with other PRs, so we shouldn't rush to get it in because of a fear that it will bit rot. It's more important that we merge it in in a functional state.

**Comment by doom777** [comment]
@IgorMinar Thanks for that note. I have updated the code. Please review it. 
Of course you're correct that it's more important that the PR is functional, then just merged.

**Comment by doom777** [comment]
Hey, @IgorMinar  @petebacondarwin can this PR move forward?

**Comment by doom777** [comment]
Hey guys, just another ping on this PR.

**Comment by petebacondarwin** [comment]
I'm afraid it is Thanksgiving in US - so it is unlikely anything will move on this before Tuesday.

**Comment by doom777** [comment]
Hey, everyone. It's Tuesday here. Just pinging.

**Review by IgorMinar — APPROVED** [review]
Lgtm now. Thank you for making all the follow up changes.

**Comment by IgorMinar** [comment]
merge-assistance: global approval

**Comment by mhevery** [comment]
[presubmit](http://test/OCL:283568912:BASE:283568958:1575396586976:aedc6e90)

**Pull request closed without merging**